martin-mbtiles/src/tile_copier.rs

```diff
@@ -144,207 +145,278 @@ impl TileCopier {
     }
 
     pub async fn run(self) -> MbtResult<SqliteConnection> {
-        let mut mbtiles_type = open_and_detect_type(&self.src_mbtiles).await?;
-        let force_simple = self.options.force_simple && mbtiles_type != TileTables;
+        let src_mbttype = open_and_detect_type(&self.src_mbtiles).await?;
 
-        let opt = SqliteConnectOptions::new()
-            .create_if_missing(true)
-            .filename(&self.options.dst_file);
-        let mut conn = SqliteConnection::connect_with(&opt).await?;
+        let mut conn = SqliteConnection::connect_with(
+            &SqliteConnectOptions::new()
+                .create_if_missing(true)
+                .filename(&self.options.dst_file),
+        )
+        .await?;
 
         let is_empty = query!("SELECT 1 as has_rows FROM sqlite_schema LIMIT 1")
             .fetch_optional(&mut conn)
             .await?
             .is_none();
 
-        if !is_empty && self.options.diff_with_file.is_some() {
+        let dst_mbttype = if is_empty {
+            let dst_mbttype = self
+                .options
+                .dst_mbttype
+                .clone()
+                .unwrap_or_else(|| src_mbttype.clone());
+
+            self.create_new_mbtiles(&mut conn, &dst_mbttype, &src_mbttype)
+                .await?;
+
+            dst_mbttype
```

Comment:
```suggestion
            self
                .options
                .dst_mbttype
                .clone()
                .unwrap_or_else(|| src_mbttype.clone());

            self.create_new_mbtiles(&mut conn, &dst_mbttype, &src_mbttype)
                .await?
```

Reply:
`dst_mbttype` is an argument for `create_new_mbtiles`

Reply:
yes - and i am not deleting it  - its declared a few lines above

Reply:
dope, silly me, thx

Reply:
but it is not assigned until the code block returns on line 172